From 87638535548f7c6ad473b582c58a8d55990fdcbf Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:26 -0400
Subject: [tools] perf/trace: Warn the user when not available

Message-id: <1410181752-25631-262-git-send-email-jolsa@redhat.com>
Patchwork-id: 91934
O-Subject: [PATCH RHEL7.1 BZ1134356 261/307] perf trace: Warn the user when not available
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 1b5726220fe16ac38eb2db43e7bc82e69f449fca
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Mon May 26 16:02:29 2014 -0300

description
===========
When the audit-libs devel package is not found at build time we disable
the 'trace' command, as we are not able to map syscall numbers to
strings, but then the message the user is presented is cryptic:

  [root@zoo linux]# trace ls
  perf: 'ls' is not a perf-command. See 'perf --help'.

Fix it by presenting a more helpful message:

  [root@zoo linux]# trace l
  trace command not available: missing audit-libs devel package at build time.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/perf.c b/tools/perf/perf.c
index 431798a..78f7b92 100644
--- a/tools/perf/perf.c
+++ b/tools/perf/perf.c
@@ -481,14 +481,18 @@ int main(int argc, const char **argv)
   fprintf(stderr, "cannot handle %s internally", cmd);
   goto out;
  }
-#ifdef HAVE_LIBAUDIT_SUPPORT
  if (!prefixcmp(cmd, "trace")) {
+#ifdef HAVE_LIBAUDIT_SUPPORT
   set_buildid_dir();
   setup_path();
   argv[0] = "trace";
   return cmd_trace(argc, argv, NULL);
- }
+#else
+  fprintf(stderr,
+   "trace command not available: missing audit-libs devel package at build time.\n");
+  goto out;
 #endif
+ }
  /* Look for flags.. */
  argv++;
  argc--;
-- 
1.7.1